packages/x-data-grid/src/components/GridSkeletonLoadingOverlay.tsx

```diff
@@ -0,0 +1,141 @@
+import * as React from 'react';
+import Skeleton from '@mui/material/Skeleton';
+
+import { styled } from '@mui/system';
+import { unstable_useForkRef as useForkRef } from '@mui/utils';
+import { useGridApiContext } from '../hooks/utils/useGridApiContext';
+import {
+  gridColumnPositionsSelector,
+  gridColumnsTotalWidthSelector,
+  gridDimensionsSelector,
+  gridVisibleColumnDefinitionsSelector,
+  useGridApiEventHandler,
+  useGridSelector,
+} from '../hooks';
+import { GridColType, GridEventListener } from '../models';
+import { createRandomNumberGenerator } from '../utils/utils';
+
+const DEFAULT_COLUMN_WIDTH_RANGE = [40, 80] as const;
+
+const COLUMN_WIDTH_RANGE_BY_TYPE: Partial<Record<GridColType, [number, number]>> = {
+  number: [40, 60],
+  string: [40, 80],
+  date: [40, 60],
+  dateTime: [60, 80],
+  singleSelect: [40, 80],
+} as const;
+
+const colWidthVar = (index: number) => `--colWidth-${index}`;
+
+const SkeletonOverlay = styled('div')({
```

Comment:
I assume every single styled should be named, e.g. https://github.com/mui/mui-x/blob/a4c6f4b8502923a453e2ccfce5361a0ab1b62a35/packages/x-data-grid/src/components/virtualization/GridVirtualScrollerRenderZone.tsx#L26

Unless there is strong reason for it to be private. This is so developers can more easily customize those. Same with having public class names.

Reply:
Makes sense, fixed here https://github.com/mui/mui-x/pull/13293/commits/721e78a55ba31395fdfb1de83616bdcfa53fb9c2#diff-9ab81d81f6b1b50b81d67721cab4a8abb428efe4839cce6856d551c4805720afR23-R26